commercetools-models/src/main/java/io/sphere/sdk/shoppinglists/LineItem.java

```diff
@@ -0,0 +1,54 @@
+package io.sphere.sdk.shoppinglists;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import io.sphere.sdk.annotations.ResourceValue;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.models.Reference;
+import io.sphere.sdk.products.ProductVariant;
+import io.sphere.sdk.producttypes.ProductType;
+import io.sphere.sdk.types.CustomFields;
+import io.sphere.sdk.types.TypeDraft;
+
+import javax.annotation.Nullable;
+import java.time.ZonedDateTime;
+
+@JsonDeserialize(as = LineItemImpl.class)
+@ResourceValue
+public interface LineItem {
+    String getId();
+
+    String getProductId();
+
+    @Nullable
+    Integer getVariantId();
+
+    @Nullable
+    Reference<ProductType> getProductType();
```

Comment:
no it's not optional.
It's optional on carts because we had to stay backwards compatible with the existing persisted data.
For a green field endpoint, I choose to have this field non optional because there is no reason for this.